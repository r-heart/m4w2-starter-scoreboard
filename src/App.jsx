import Display from "./routes/Display";
import Setup from "./routes/Setup";

function App() {
  return (
    // using shorthand for a React Fragment
    <>
      <h1 className="my-4 text-center text-3xl font-extrabold">
        Scoreboard App
      </h1>
      <main>
        <Display />
        <Setup />
      </main>
    </>
  );
}

export default App;
