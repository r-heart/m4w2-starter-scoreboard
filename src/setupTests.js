import "@testing-library/jest-dom";
import { cleanup, render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { afterEach } from "vitest";

afterEach(() => {
  cleanup();
});

export function setup(component) {
  return {
    user: userEvent.setup(),
    ...render(component),
  };
}
