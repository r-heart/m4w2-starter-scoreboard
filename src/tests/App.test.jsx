import { render, screen, waitFor } from "@testing-library/react";
import { setup } from "src/setupTests.js";
import App from "../App";
import CONFIG from "../config";

test("intitial render", async () => {
  const rendered = render(<App />);
  expect(rendered).toMatchSnapshot();
});

it("renders the correct buttons whenever a sport is selected", async () => {
  const { user } = setup(<App />);

  const select = screen.getByRole("combobox");

  // Select the first actual choice (not the placeholder option)
  await user.selectOptions(select, "⚽/🏒");

  const buttons = screen.getAllByRole("button");

  expect(buttons).toHaveLength(CONFIG[0].buttons.length);
});

it("limits the numbers of periods to the number specified in the config", async () => {
  const { user } = setup(<App />);

  const periodsInput = screen.getByLabelText(/periods/i);
  const goBtn = screen.getByRole("button", { name: /Go/i });

  await user.type(periodsInput, "3");
  await user.click(goBtn);

  const nextPeriodBtn = await screen.findByRole("button", {
    name: "Next Period",
  });
  const periodP = await screen.findByTestId("period");

  await user.click(nextPeriodBtn);
  await user.click(nextPeriodBtn);
  await user.click(nextPeriodBtn);

  expect(periodP).toHaveTextContent("3");
});

it("starts with the correct time remaining", async () => {
  const { user } = setup(<App />);

  const timeRemainingInput = screen.getByLabelText(/time/i);
  const goBtn = screen.getByRole("button", { name: /Go/i });

  await user.type(timeRemainingInput, "10");
  await user.click(goBtn);

  const startBtn = await screen.findByRole("button", { name: "Start" });
  const timeDisplay = await screen.findByTestId("time");
  await user.click(startBtn);

  await waitFor(() => {
    expect(timeDisplay).toHaveTextContent("10:00");
  });
});
