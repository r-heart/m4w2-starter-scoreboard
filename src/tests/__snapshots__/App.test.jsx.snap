// Vitest Snapshot v1

exports[`intitial render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1
        class="my-4 text-center text-3xl font-extrabold"
      >
        Scoreboard App
      </h1>
      <main
        class="flex flex-col items-center gap-y-4"
      >
        <label
          class="sr-only"
          for="sport-select"
        >
          Choose a pet:
        </label>
        <select
          id="sport-select"
        >
          <option
            value=""
          >
            --Please choose your sport--
          </option>
          <option
            value="‚öΩ/üèí"
          >
            ‚öΩ/üèí
          </option>
          <option
            value="üèÄ"
          >
            üèÄ
          </option>
          <option
            value="üèà"
          >
            üèà
          </option>
        </select>
        <label
          class="sr-only"
          for="periods"
        >
          Number of Periods
        </label>
        <input
          class="rounded-md border border-gray-300 p-2"
          id="periods"
          placeholder="Number of Periods"
          type="number"
          value=""
        />
        <label
          class="sr-only"
          for="time"
        >
          Time per period? (minutes)
        </label>
        <input
          class="rounded-md border border-gray-300 p-2"
          id="time"
          placeholder="Time per period? (minutes)"
          type="number"
          value=""
        />
        <button
          class="rounded bg-green-500 px-4 py-2 text-white"
          data-count="Go!"
        >
          Go!
        </button>
      </main>
    </div>
  </body>,
  "container": <div>
    <h1
      class="my-4 text-center text-3xl font-extrabold"
    >
      Scoreboard App
    </h1>
    <main
      class="flex flex-col items-center gap-y-4"
    >
      <label
        class="sr-only"
        for="sport-select"
      >
        Choose a pet:
      </label>
      <select
        id="sport-select"
      >
        <option
          value=""
        >
          --Please choose your sport--
        </option>
        <option
          value="‚öΩ/üèí"
        >
          ‚öΩ/üèí
        </option>
        <option
          value="üèÄ"
        >
          üèÄ
        </option>
        <option
          value="üèà"
        >
          üèà
        </option>
      </select>
      <label
        class="sr-only"
        for="periods"
      >
        Number of Periods
      </label>
      <input
        class="rounded-md border border-gray-300 p-2"
        id="periods"
        placeholder="Number of Periods"
        type="number"
        value=""
      />
      <label
        class="sr-only"
        for="time"
      >
        Time per period? (minutes)
      </label>
      <input
        class="rounded-md border border-gray-300 p-2"
        id="time"
        placeholder="Time per period? (minutes)"
        type="number"
        value=""
      />
      <button
        class="rounded bg-green-500 px-4 py-2 text-white"
        data-count="Go!"
      >
        Go!
      </button>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
